scriptmodules/archivefuncs.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+## @file archivefuncs.sh
+## @brief RetroPie archivefuncs library
+## @copyright GPLv3
+
+readonly arch_dir="/tmp/retropie_archive"
+
+function __zipExtract() {
+    unzip "$1" -d "$2"
+}
+
+## @fn archiveExtract()
+## @param src_file Full path to archive file to extract
+## @param disk_exts Comma seperated list of supported disk extensios
+## @brief Extracts an archive to a temporary directory
+## @details After calling this the variable arch_dir will contain the directory that was used 
+## for extraction. Also the variable arch_files will contain an array of filenames that are 
+## considered as game disks according to parameter disk_exts.
+function archiveExtract() {
+    local src_file=$1
+    local disk_exts=$2
+
+    # clean temp directory if needed
+    archiveCleanup
+    mkdir "$arch_dir"
+
+    local ext="${src_file##*.}"    
+    if [[ ${ext,,} == "zip" ]]; then
+        __zipExtract $src_file $arch_dir
+    else
+        echo "Unsupported archive: $src_file"
+        return 1
+    fi
+
+    # get a list of extracted files matching the desired extensions
+    shopt -s nocaseglob nullglob
+    read -r -a arch_files <<< $(eval "ls -1 $arch_dir/*{$disk_exts,}")
```

Comment:
Why eval ?

I would think `find` would be better choice, and you wouldn't need to change the shell opts. I suspect this may have problems with filenames with spaces ?